tests/foreman/cli/test_repository.py

```diff
@@ -2186,7 +2189,28 @@ def test_negative_update_red_hat_repo():
 
         :expectedresults: hammer returns error code. The repository is not updated.
         """
-        pass
+
+        rh_repo_set_id = RepositorySet.list({'organization-id': module_manifest_org.id})[0]['id']
+
+        RepositorySet.enable(
+            {
+                'organization-id': module_manifest_org.id,
+                'basearch': "x86_64",
+                'id': rh_repo_set_id,
+            }
+        )
+        repo_list = Repository.list({'organization-id': module_manifest_org.id})
+
+        rh_repo_id = Repository.list({'organization-id': module_manifest_org.id})[0]['id']
+
+        Repository.update(
+            {
+                'id': rh_repo_id,
+                'url': f'{gen_url(scheme="https")}:{gen_integer(min_value=10, max_value=9999)}',
+            }
+        )
+        repo_info = Repository.info({'organization-id': module_manifest_org.id, 'id': rh_repo_id})
+        assert repo_info['url'] in [i.get('url') for i in repo_list]
```

Comment:
non-blocking, instead of `i` good practice to use valid naming `repo`.  
```suggestion
        assert repo_info['url'] in [repo.get('url') for repo in repo_list]
```